Dockerfile

```diff
@@ -33,31 +42,34 @@ COPY start_backup.sh /start_backup.sh
 COPY python/delete_instuck_progress.py /delete-instuck-backups/delete_instuck_progress.py
 
 # copy cron job
-COPY cron-ghe-backup-automata /etc/cron.d/ghe-backup
+COPY cron-ghe-backup /etc/cron.d/ghe-backup
 
 # copy finale CMD commands
 COPY final-docker-cmd.sh /backup/final-docker-cmd.sh
 
+
 #PLACEHOLDER_4_COPY_SCM_SOURCE_JSON
 
-# change mode of files
 RUN \
+# change mode of files
   chown -R application: /data && \
   chown -R application: /backup && \
   chown -R application: /kms && \
   chown -R application: /delete-instuck-backups && \
-  chown -R root: /start_backup.sh && \
+  chown -R application: /start_backup.sh && \
   chmod 0700 /kms/extract_decrypt_kms.py && \
   chmod 0700 /kms/convert-kms-private-ssh-key.sh && \
   chmod 0644 /etc/cron.d/ghe-backup && \
   chmod 0700 /delete-instuck-backups/delete_instuck_progress.py && \
   chmod 0700 /start_backup.sh && \
   chmod 0700 /backup/final-docker-cmd.sh && \
-  mkfifo /var/log/ghe-prod-backup.log
-
-# delete_instuck_progress log
-RUN \
+  mkfifo /var/log/ghe-prod-backup.log && \
+  chown -R application: /var/log/ghe-prod-backup.log && \
   touch /var/log/ghe-delete-instuck-progress.log && \
   chown -R application: /var/log/ghe-delete-instuck-progress.log
 
-CMD ["/backup/final-docker-cmd.sh"]
+USER application
+
+# https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/#user mentions to avoid sudo,
+# however cron as part of the final-docker-cmd.sh has to run as
+CMD ["sudo", "/backup/final-docker-cmd.sh"]
```

Comment:
if you execute /backup/final-docker-cmd.sh with sudo , then /kms/convert-kms-private-ssh-key.sh will create the ssh-key in /root/.ssh folder, which will be inaccessible for application user, so ssh into ghe instance will fail:  https://github.com/zalando/ghe-backup/pull/76/files#diff-2a885483390ec330ff6c432b37530949.
Just faced this in ghe-hubble.
workaround would either using sudo in final-docker-cmd.sh (`sudo /usr/sbin/cron`), or adapting /kms/convert-kms-private-ssh-key.sh to create the ssh-key in the right place.

Reply:
Thank you, I deployed on Friday to test but did not check back the deployment results.